### ISSUE TITLE
OSX App can't update itself when installed

### ISSUE THREAD

**Issue by enisn** [velopack/velopack#185]
Update logic is perfectly working when using the portable version.
But, when the Application is installed via `.pkg` file on OSX _(Located under `/Applications` folder.)_ Delta or Full updates don't work.

The exception: 

```
An error occurred trying to start process '/Applications/AbpStudio.app/Contents/MacOS/UpdateMac' with working directory '/Applications/AbpStudio.app/Contents/MacOS'. Permission denied
System.ComponentModel.Win32Exception (13): An error occurred trying to start process '/Applications/AbpStudio.app/Contents/MacOS/UpdateMac' with working directory '/Applications/AbpStudio.app/Contents/MacOS'. Permission denied
   at System.Diagnostics.Process.ForkAndExecProcess(ProcessStartInfo startInfo, String resolvedFilename, String[] argv, String[] envp, String cwd, Boolean setCredentials, UInt32 userId, UInt32 groupId, UInt32[] groups, Int32& stdinFd, Int32& stdoutFd, Int32& stderrFd, Boolean usesTerminal, Boolean throwOnNoExec)
   at System.Diagnostics.Process.StartCore(ProcessStartInfo startInfo)
   at System.Diagnostics.Process.Start(ProcessStartInfo startInfo)
   at Velopack.UpdateExe.Apply(IVelopackLocator locator, VelopackAsset toApply, Boolean silent, Boolean restart, String[] restartArgs, ILogger logger)
   at Velopack.UpdateManager.WaitExitThenApplyUpdates(VelopackAsset toApply, Boolean silent, Boolean restart, String[] restartArgs)
   at Velopack.UpdateManager.ApplyUpdatesAndRestart(VelopackAsset toApply, String[] restartArgs)
```

Probably  starting `UpdateMac` by osascript with `do shell script ... with administrator privileges` can solve the problem. 

It's not the best way as you mentioned here https://github.com/velopack/velopack/issues/50 but for now it can fix the problem until finding another solution

---

Also demo has the same result
(https://github.com/velopack/velopack/tree/develop/samples/AvaloniaCrossPlat)

**Comment by enisn** [comment]
By the way, 
- ✅ Tt works when installed with `Install for me Only`
- ❌ It requires admin privileges when installed with `Install for all users of this computer`
![image](https://github.com/user-attachments/assets/b637bc20-436a-46b8-9bfa-5d10db1403fa)

> **Note** that, it's not about the installer, installer works perfectly in both of situations. Only the updater doesn't work when installed for all users because of permission denied.

**Comment by caesay** [comment]
This was fixed in [0.0.571](https://github.com/velopack/velopack/releases/tag/0.0.571-gc0a9476)
